Source/ProgressionSystemRuntime/Public/Widgets/PSOverlayWidget.h

```diff
@@ -13,21 +13,58 @@ UCLASS()
 class PROGRESSIONSYSTEMRUNTIME_API UPSOverlayWidget : public UUserWidget
 {
 	GENERATED_BODY()
+
 public:
-	
+	/**
+	* Sets the visibility of the overlay elements and playing fade animation if needed 
+	* @param VisibilitySlate The visibility state (e.g., Visible, Collapsed) to apply to the overlay and icon.
+	* @param bShouldPlayFadeAnimation Defines if the fade animation should be played after or before widget is visible
+	*/
+	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
+	void SetOverlayVisibility(ESlateVisibility VisibilitySlate, bool bShouldPlayFadeAnimation);
+
+protected:
+	/** overrides NativeTick to make the user widget tickable **/ 
+	virtual void NativeTick(const FGeometry& MyGeometry, float InDeltaTime) override;
+	/** Event to execute when widget is ready */
+	virtual void NativeConstruct() override;
+
+	/**
+	* Play the overlay elemetns fade-in/fade-out animation.
+	* Uses the internal FadeCurveFloatInternal initialized in NativeConstruct
+	* @param StartTimeRef The start timer to calculate time elapsed
+	*/
+	bool FadeOverlayElementsAnimation(float& StartTimeRef);
+
 	/**
 	* Sets the visibility of the background overlay and lock icon.
 	* @param VisibilitySlate The visibility state (e.g., Visible, Collapsed) to apply to the overlay and icon.
 	*/
 	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
-	void SetOverlayVisibility(ESlateVisibility VisibilitySlate);
+	void SetOverlayItemsVisibility(ESlateVisibility VisibilitySlate);
 	
-protected:
 	/** Background overlay for tint effect */
 	UPROPERTY(EditDefaultsOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, BindWidget))
 	TObjectPtr<class UImage> PSCBackgroundOverlay = nullptr;
 
 	/** Background overlay lock icon */
 	UPROPERTY(EditDefaultsOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, BindWidget))
 	TObjectPtr<class UImage> PSCBackgroundIconLock = nullptr;
+
+	/** Stores the starting time to fade-in/fade-out overlay in the main menu when cinematic started */
+	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Starting time to hide stars"))
+	float StartTimeFadeAnimationInternal = 0.0f;
+
+	/** if the fade-in/fade-out overlay animation in the main menu when cinematic started should be played */
+	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Starting time to hide stars"))
+	bool bShouldPlayFadeAnimationInternal = false;
```

Comment:
I hardly understand why you need both class properties `bIsFadeInAnimationInternal` and `bShouldPlayFadeAnimationInternal` at the same time. You already have `StartTimeFadeAnimationInternal` which if zero means animation should not playing. 

Even their duplicated comment highlights one of booleans are not needed.

Reply:
This is to support animation or skip. In case you would need to execute same but without fade animation for some reason

Reply:
@h4rdmol, in original comment, I meant you have unnecessary transient properties here, like bIsFadeInAnimationInternal which can be just removed without any impact on the logic, please see [this comment](https://github.com/h4rdmol/ProgressionSystem/pull/20/files#r1805857435), where I show example of code without `bIsFadeInAnimationInternal` usage

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/20/commits/b31977e4e0c4f1a106d0c5d359e60d669bc2e155